### PR TITLE
[#9492] Fix twisted.python.rebuild on Python 3.7

### PR DESCRIPTION
**Pull request by hawkowl**
https://twistedmatrix.com/trac/ticket/9492

### CODE REVIEW

**Comment by hawkowl** [comment]
Fixed on the 18.04-3.7 builder ^

File: src/twisted/python/rebuild.py [hunk 1]
```diff
@@ -150,7 +150,7 @@ def updateInstance(self):
 
 
 
-def __getattr__(self, name):
+def __injectedgetattr__(self, name):
```
Comment (glyph) [inline src/twisted/python/rebuild.py:153]: zomg hilaaaaaaaaaaaarious

File: docs/historic/index.rst [hunk 2]
```diff
@@ -6,6 +6,11 @@
 Historical Documents
 ====================
 
+.. toctree::
```
Comment (glyph) [inline docs/historic/index.rst:9]: sneaky

I'll allow it

File: docs/historic/NEWS.rst [hunk 3]
```diff
@@ -2060,7 +2060,7 @@ Bugfixes
 
 Deprecations and Removals
 -------------------------
- - The --user option to `twistd manhole' has been removed as it was
+ - The --user option to ``twistd manhole`` has been removed as it was
```
Comment (glyph) [inline docs/historic/NEWS.rst:2063]: lol

File: docs/historic/NEWS.rst [hunk 4]
```diff
@@ -937,7 +937,7 @@ Bugfixes
    values of KEYWORD and UNKEYWORD and also emits them without adding
    quotes (which are also illegal). (#4392)
  - twisted.mail.imap4.IMAP4Client can now interpret the BODY response
-   for multipart/* messages with parts which are also multipart/*.
+   for ``multipart/*`` messages with parts which are also ``multipart/*``.
```
Comment (glyph) [inline docs/historic/NEWS.rst:940]: lollll

File: docs/historic/NEWS.rst [hunk 5]
```diff
@@ -2248,7 +2248,7 @@ Features
    st_dev, st_nlink, st_uid, and st_gid fields.  This is in
    preparation for the deprecation of FilePath.statinfo. (#4712)
  - IPv4Address and UNIXAddress now have a __hash__ method. (#4783)
- - twisted.protocols.ftp.FTP.ftp_STOR now catches `FTPCmdError`s
+ - twisted.protocols.ftp.FTP.ftp_STOR now catches ``FTPCmdError`` s
```
Comment (glyph) [inline docs/historic/NEWS.rst:2251]: erm, do we have to write news files in ReST now?

**Review by glyph — APPROVED** [review]
i lold

File: docs/historic/NEWS.rst (hunk 5 above)
Comment (hawkowl) [reply docs/historic/NEWS.rst:2251]: considering the file format is ".rst", I am guessing that writing them in rest compatible syntax is a good idea, yeah

File: src/twisted/python/rebuild.py (hunk 1 above)
Comment (hawkowl) [reply src/twisted/python/rebuild.py:153]: https://www.python.org/dev/peps/pep-0562/

**Pull request merged**